torch/fx/passes/split_module.py

```diff
@@ -207,14 +279,22 @@ def instantiate_node_partition_mapping(node):
             )
             continue
 
-        instantiate_node_partition_mapping(node)
+        if assert_monotonically_increasing:
```

Comment:
this is to enforce the graph is split into sequential chunks of nodes, rather than mapping some 'earlier' nodes to 'later' partitions?

Reply:
Yes. Without this property we cannot be assured about which partitions will covered by a given enter and exit.